### DISCUSSION TITLE
Export Problem

### DISCUSSION THREAD

**Discussion by gasmasks** [lairdubois/lairdubois-opencutlist-sketchup-extension#489] [Q&A]
Hello Hubbers

I have a request

when i export csv, when there is no edge in a part, it shows two double quotations ("")
for example
45.91;1.90;0.20;"";"";"";"";"";"";"";""
45.91;1.90;0.20;"";(0.2x1.9);(0.2x1.9);(0.2x1.9);(0.2x1.9);"";"";""

i want to prevent double quotations from showing up

is there any solution in xml coding or else
plz help

thanks

**Comment by bbeaulant** [comment]
Hi,
This is the default behavior of the Ruby CSV writer to informs the futur reader that the empty value is a string. This will not change.
A sort of workaround could be to customize the exported value with ruby code in OpenCutList to returns a "space" when edge is empty.

<img width="709" alt="Capture d’écran 2023-02-27 à 11 36 26" src="https://user-images.githubusercontent.com/156056/221541504-f873041b-4bb7-4f25-81a7-f9d1bf1c7e54.png">

**Comment by gasmasks** [comment]
Hello @bbeaulant
Thanks so much for your clear answer
The problem is i want to export to opticut
For the last question, is there any solution in the future? Should i wait?

**Reply by bbeaulant** [reply]
And what is the problem with importing in opticut ?

**Reply by gasmasks** [reply]
opticut cant understand double quotes
if double quotes replaced with SPACE or Empty, its Done

**Reply by bbeaulant** [reply]
Poor OptiCut :)

> if double quotes replaced with SPACE or Empty, its Done

Then problem solved, no ?

**Reply by gasmasks** [reply]
yeah
i opened csv file inside notepad and find and replaced all ("")
and it solved

**Reply by gasmasks** [reply]
but it would be better to be done automatically by opencutlist
i hope so

**Reply by bbeaulant** [reply]
Hmm, this was what my first answer say. You can.

**Reply by bbeaulant** [reply]
You just need to customize the output value for each edge.

<img width="709" alt="Capture d’écran 2023-02-27 à 11 36 26" src="https://user-images.githubusercontent.com/156056/221541347-23b1dc20-9d0d-4e3c-828c-adc4aa93b7bf.png">

**Reply by gasmasks** [reply]
testing right now

**Reply by bbeaulant** [reply]
It will be never "solved" on OCL side, because the main problem is certainly because OptiCut CSV Reader do not follow the standards ...

**Reply by gasmasks** [reply]
THanks BRO
its done
now its working great

i wish you all the best to come to your life
god bless you

**Reply by gasmasks** [reply]
and by the way
is there any way to export drillings from sketchup

**Reply by bbeaulant** [reply]
Not with OpenCutList.

**Reply by gasmasks** [reply]
can u plz tell me a name of plugin in sketchup to export drillings
thanks

**Comment by boreshkar** [comment]
Hello Friends

what is the problem with this code?

..........................................................................
if @ Edge Lenght1."P (0.2 x 1.6 cm)"
"P"
else
""
end
..........................................................................

the point is to remove  **(0.2 x 1.6 cm)**
and **P** Remains

**Reply by bbeaulant** [reply]
Like the docs says ([link](https://docs.opencutlist.org/features/parts/export-to-csv/customization#formulas)), `@edges` variables as 3 sub properties : 

<img width="522" alt="Capture d’écran 2024-12-10 à 16 27 50" src="https://github.com/user-attachments/assets/c9db7fbf-35bc-4a7d-b786-608646d27e70">

So if you want only the material name, just type `@edge_ymin.material_name`

**Reply by boreshkar** [reply]
thakns bro
it made me so happy
